### ISSUE TITLE
False positives with Svelte 5 event handlers and component library props (bits-ui)

### ISSUE THREAD

**Issue by lukaskawerau** [pheuter/svelte-check-rs#76]
## Description

svelte-check-rs reports false positives for valid Svelte 5 syntax, particularly:

1. **Event handlers as props** (`onclick`, `onkeydown`, etc.)
2. **Component library props** (e.g., `variant` on bits-ui components)
3. **SvelteKit `$props()` pattern** ("Variable 'data' is used before being assigned")

## Environment

- svelte-check-rs: 0.6.9
- svelte: 5.47.1
- @sveltejs/kit: 2.16.0
- bits-ui: 1.4.3
- Node: 22.12.0
- OS: Linux x86_64

## Reproduction

1. Create a SvelteKit project with Svelte 5
2. Use `onclick={handler}` syntax (new Svelte 5 event syntax)
3. Use bits-ui components with props like `variant`
4. Run `svelte-check-rs --tsconfig ./tsconfig.json`

## Expected

No errors (same as `svelte-check`)

## Actual

```
src/routes/settings/+page.svelte:388:15
Error: Object literal may only specify known properties, and 'onclick' does not exist in type '{} & __SvelteCssProps'. (ts(TS2353))

src/routes/settings/credits/+page.svelte:99:13
Error: Object literal may only specify known properties, and 'variant' does not exist in type '{} & __SvelteCssProps'. (ts(TS2353))

src/routes/vision/new/+page.svelte:8:22
Error: Variable 'data' is used before being assigned. (ts(TS2454))
```

Running `svelte-check` on the same codebase reports 0 errors.

## Minimal example

```svelte
<script lang="ts">
  let { data } = $props();  // Error: 'data' used before assigned
</script>

<button onclick={() => console.log('clicked')}>  <!-- Error: 'onclick' does not exist -->
  Click me
</button>
```

## Notes

The errors suggest the Svelte 5 → TypeScript transformation isn't correctly handling:
- The new event handler syntax (`onclick` vs `on:click`)
- Component library type definitions
- SvelteKit's `$props()` destructuring pattern

**Comment by lukaskawerau** [comment]
You're right - this was caused by stray Svelte 4 style artifacts in my original codebase. Sorry for the trouble!
